gre/config.ts

```diff
@@ -0,0 +1,223 @@
+import fs from 'fs'
+import path from 'path'
+
+import { NetworkConfig, NetworksConfig } from 'hardhat/types/config'
+import { HardhatRuntimeEnvironment } from 'hardhat/types/runtime'
+import { HttpNetworkConfig } from 'hardhat/types/config'
+
+import { GraphRuntimeEnvironmentOptions } from './type-extensions'
+import { GREPluginError } from './helpers/error'
+import GraphNetwork from './helpers/network'
+
+import { createProvider } from 'hardhat/internal/core/providers/construction'
+import { EthersProviderWrapper } from '@nomiclabs/hardhat-ethers/internal/ethers-provider-wrapper'
+
+import { logDebug } from './logger'
+
+interface GREChains {
+  l1ChainId: number
+  l2ChainId: number
+  isHHL1: boolean
+  isHHL2: boolean
+}
+
+interface GREProviders {
+  l1Provider: EthersProviderWrapper | undefined
+  l2Provider: EthersProviderWrapper | undefined
+}
+
+interface GREGraphConfigs {
+  l1GraphConfigPath: string | undefined
+  l2GraphConfigPath: string | undefined
+}
+
+export function getAddressBookPath(
+  hre: HardhatRuntimeEnvironment,
+  opts: GraphRuntimeEnvironmentOptions,
+): string {
+  logDebug('== Getting address book path')
+  logDebug(`Graph base dir: ${hre.config.paths.graph}`)
+  logDebug(`1) opts.addressBookPath: ${opts.addressBook}`)
+  logDebug(`2) hre.config.graph.addressBook: ${hre.config.graph?.addressBook}`)
+
+  let addressBookPath = opts.addressBook ?? hre.config.graph?.addressBook
+
+  if (addressBookPath === undefined) {
+    throw new GREPluginError('Must set a an addressBook path!')
+  }
+
+  addressBookPath = normalizePath(addressBookPath, hre.config.paths.graph)
+
+  if (!fs.existsSync(addressBookPath)) {
+    throw new GREPluginError(`Address book not found: ${addressBookPath}`)
+  }
+
+  logDebug(`Address book path found: ${addressBookPath}`)
+  return addressBookPath
+}
+
+export function getChains(mainChainId: number | undefined): GREChains {
+  logDebug('== Getting chain ids')
+  logDebug(`Hardhat chain id: ${mainChainId}`)
+
+  if (!GraphNetwork.isSupported(mainChainId)) {
+    throw new GREPluginError(`Chain ${mainChainId} is not supported!`)
+  }
+
+  // If mainChainId is supported there is a supported counterpart chainId so both chains are not undefined
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  mainChainId = mainChainId!
+
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const secondaryChainId = GraphNetwork.counterpart(mainChainId)!
+  logDebug(`Secondary chain id: ${secondaryChainId}`)
+
+  const isHHL1 = GraphNetwork.isL1(mainChainId)
+  const isHHL2 = GraphNetwork.isL2(mainChainId)
+  const l1ChainId = isHHL1 ? mainChainId : secondaryChainId
+  const l2ChainId = isHHL2 ? mainChainId : secondaryChainId
+
+  logDebug(`L1 chain id: ${l1ChainId} - Is HHL1: ${isHHL1}`)
+  logDebug(`L2 chain id: ${l2ChainId} - Is HHL2: ${isHHL2}`)
+
+  return {
+    l1ChainId,
+    l2ChainId,
+    isHHL1,
+    isHHL2,
```

Comment:
Is the isHHL2 ever used? as the first isHHL1 is binary. Or this is just included for symmetry?

Reply:
No, `isHHL2` is not used. Just included as you say for symmetry. Perhaps we can remove it and rename the `isHHL1` variable to something else but I struggled to find a good name.